src/algorithms/synthesis/syrec_synthesis.cpp

```diff
@@ -584,7 +666,73 @@ namespace syrec {
         return true;
     }
 
-    bool standard_syrec_synthesizer::on_statement(const call_statement& statement) {
+    bool SyrecSynthesisNoAdditionalLines::on_statement(const for_statement& statement) {
+        const auto& [nfrom, nto] = statement.range;
+
+        const unsigned     from          = nfrom ? nfrom->evaluate(SyrecSynthesis::loop_map) : 1u; // default value is 1u
+        const unsigned     to            = nto->evaluate(SyrecSynthesis::loop_map);
+        const unsigned     step          = statement.step ? statement.step->evaluate(loop_map) : 1u; // default step is +1
+        const std::string& loop_variable = statement.loop_variable;
+
+        if (from <= to) {
+            for (unsigned i = from; i <= to; i += step) {
+                // adjust loop variable if necessary
+
+                if (!loop_variable.empty()) {
+                    loop_map[loop_variable] = i;
+                }
+
+                for (const auto& stat: statement.statements) {
+                    if ((!full_statement(stat)) && (!on_statement(stat)))
+                        return false;
+                }
+            }
+        }
+
+        else if (from > to) {
+            for (auto i = (int)from; i >= (int)to; i -= (int)step) {
+                // adjust loop variable if necessary
+
+                if (!loop_variable.empty()) {
+                    loop_map[loop_variable] = i;
+                }
+
+                for (const auto& stat: statement.statements) {
+                    if ((!full_statement(stat)) && (!on_statement(stat)))
+                        return false;
+                }
+            }
+        }
+        // clear loop variable if necessary
+        if (!loop_variable.empty()) {
+            assert(loop_map.erase(loop_variable) == 1u);
+        }
+
+        return true;
+    }
+
+    bool SyrecSynthesisAdditionalLines::on_statement(const call_statement& statement) {
```

Comment:
These functions are also identical in both derived classes

Reply:
the onstatement used in these functions are not the same